models/builtin_models.h

```diff
@@ -0,0 +1,28 @@
+/*
+ *  builtin_models.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace nest
+{
+
+void register_builtin_models();
```

Comment:
Should it be in its own namespace (under the ``nest`` namespace)?

Reply:
I don't think so. It's just a naked function that belongs to NEST.

Reply:
I am also a bit uncertain about the naming scheme. In one way, the scheme proposed is clear: all that is in `models` is built-in and nothing else can be built-in. But how will this scheme generalize? We might at some point soon have a large part of models auto-generated by NESTML and a few still based on hand-written code. I'd expect that there would be one models directory for each "source". And we need a clear naming for the corresponding extension module function as well. I think a naming scheme that can handle all situations consistently would be good to have.